app/stats/Metrics.tsx

```diff
@@ -96,13 +98,38 @@ const MetricBox = ({
     data: samples.map((s) => s.label),
     scaleType: "band",
   };
+
+  const calculateTrend = (data: Array<number>) => {
```

Comment:
Tohle skončí chybou, když bude ten vzorek jenom jeden. Doporučuju se u těch algoritmů vždycky zamyslet nad patologickými případy. (_A software tester walks into a bar. Orders a beer. 2 beers. 0 beers. –1 beer. 99999999 beers. "khfkew" beers. A lizard in a beer glass._ )


Obecně je to samotné počítání trendu dobrý kandidát pro vytažení do `src/data/metrics.ts` a napsání pár testů? Kdybys chtěla, můžu ukázat. Dovedu si představit, že bys tam měla například funkci:

```typescript
export type Trend = "better" | "worse" | "same";
export function calculateTrend(metric: MetricDefinition, samples: MetricSample[]): [number, Trend] {…}
```

Takže by volající jen dodal metriku a vzorky a zjistil by, jaký je poslední trend a jestli je dobrý, špatný, nebo zhruba stejný. A to i pro ty „patologické“ případy (žádné vzorky, jeden vzorek).

PS. Místo `Array<number>` se většinou idiomaticky píše `number[]`.